cx/telegram/server.js

```diff
@@ -31,64 +32,78 @@ const {SessionsClient} = require('@google-cloud/dialogflow-cx');
 
-/**
- * Converts Telgram request to a detectIntent request.
- */
+// Converts Telgram request to a detectIntent request.
 function telegramToDetectIntent(telegramRequest, sessionPath){
-
-  const request = {
-    session: sessionPath,
-    queryInput: {
-      text: {
-        text: telegramRequest.message.text,
-      },
-      languageCode,
-    },
-  };
-
-  return request;
-}
-
-/**
- * Converts detectIntent response to a Telegram text message request. 
- */
-function detectIntentToTelegramMessage(response,chatId){
-  agentResponse = '';
-    
-  for (const message of response.queryResult.responseMessages) {
-    if (message.text) {
-      agentResponse += `${message.text.text}\n`;
-    };
-  };
-    
-  if(agentResponse.length != ''){
     const request = {
-      chat_id: chatId,
-      text: agentResponse
+        session: sessionPath,
+        queryInput: {
+            text: {
+                text: telegramRequest.message.text,
+            },
+            languageCode,
+        }
     };
+
     return request;
-  };
-};
+}
+
+// Converts detectIntent responses to Telegram message requests.
+async function convertToTelegramMessage(responses, chatId) {
+    let replies = [];
+
+    for(let response of responses.queryResult.responseMessages) {
+        let reply;
+
+        switch(true) {
+            case response.hasOwnProperty('text'): {
+                reply = {
+                    chat_id: chatId,
+                    text: response.text.text.join()
+                };  
+            }
+            break;
+
+            /**
+             * The layout for the custom payload responses can be found in these sites:
+             * Buttons: https://core.telegram.org/bots/api#inlinekeyboardmarkup
+             * Photos: https://core.telegram.org/bots/api#sendphoto
+             * Voice Audios: https://core.telegram.org/bots/api#sendvoice
+             */
+            case response.hasOwnProperty('payload'): {
+                reply = await structProtoToJson(response.payload)
+                reply['chat_id'] = chatId
+            }
+            break;
+
+            default:
+        }
+        if (reply) {
+            replies.push(reply);
+        }
+    }
+
+    return replies;
+}
 
 /**
  * Takes as input a request from Telegram and converts the request to
  * detectIntent request which is used to call the detectIntent() function
  * and finally output the response given by detectIntent().
  */
 async function detectIntentResponse(telegramRequest) {
-  const sessionId = telegramRequest.message.chat.id;
-  const sessionPath = client.projectLocationAgentSessionPath(
-    projectId,
-    locationId,
-    agentId,
-    sessionId
-  );
-  console.info(sessionPath);
-  
-  request = telegramToDetectIntent(telegramRequest, sessionPath);
-  const [response] = await client.detectIntent(request);
-
-  return response;
+    const sessionId = telegramRequest.message.chat.id;
```

Comment:
shall we observe the original 2-space indent, here and anywhere?

I see you're using 4-space indent

Reply:
I have used 4-space indent through out the integration files. The original was just autoformated to a 2-space indent by accident for a couple of lines.

Reply:
Just to confirm, there will be another PR to auto-format and fix the style?

Reply:
Yes, I will be creating another PR to format and fix the style.